### ISSUE TITLE
[Bug] `on offer: require ... 0` mission syntax does not check the player's cargo

### ISSUE THREAD

**Issue by TrebledJ** [endless-sky/endless-sky#6574]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

The `on offer: requires <outfit> 0` mission syntax doesn't work if the outfit in question is in the player's cargo (and if the mission is labelled `landing`).

### Steps to Reproduce

1. Use this mission to test.
    ```
    mission "test"
	landing
	name "Test."
	description "Test."
	to offer
		day > 16 						# To give you time to load the blaster.
	on offer
		require "Energy Blaster" 0
		conversation
			`(on offer) This shouldn't be triggered.`
				accept
	on complete
		require "Energy Blaster" 0
		conversation
			`(on complete) This shouldn't be triggered.`
    ```
2. Start a new pilot. Name and starter ship is irrelevant. Skip the tutorial.
3. Put an "Energy Blaster" into your **cargo**. Make sure it's in your cargo and not installed on the ship.
4. Depart and land. The mission is offered.
5. (Optional.) Depart and land again. The mission is now completed.

### Expected Behavior

The `requires ... 0` is not respected. The mission should only be offered when the player is not carrying a blaster.

The code for checking the required outfits takes into account every `ship->Cargo()`, but doesn't take into account `player.Cargo()` (which is where the cargo outfits are when landed).
```cpp
for(const auto &ship : player.Ships())
    if(!ship->IsDestroyed())
    {
        available += ship->Cargo().Get(it.first);
        available += ship->OutfitCount(it.first);
    }
```

### Operating System

macOS 10.15.7

### Game Source

GitHub Releases

### Game Version

898df94cf1bccc6b523cc126838cd2308f59ef3c

**Comment by tehhowch** [comment]
It does appear I missed this when refactoring in #3005

**Comment by Amazinite** [comment]
Fixed in 2625b6e6704bd83dc2bc4d07f9a6b23d23795c7e.
